mozci/task.py

```diff
@@ -82,6 +100,29 @@ class TestTask(Task):
     _errors: List = field(default=None)
     _groups: List = field(default=None)
 
+    # XXX: Once bug 1613937 and bug 1613939 are fixed, we can remove the filtering
+    # and slash replacing, and turn the warning on bad group names into an assertion.
+    def __post_init__(self):
+        if self._groups is not None:
+            self._groups = [
+                group.replace("\\", "/")
+                for group in self._groups
+                if not is_bad_group(self.id, group)
+            ]
+
+        def update_group(result):
+            if result.group is not None:
+                result.group = result.group.replace("\\", "/")
+
+            return result
+
+        if self._results is not None:
+            self._results = [
+                update_group(result)
+                for result in self._results
+                if result.group is None or not is_bad_group(self.id, result.group)
```

Comment:
Here we assert that `result.group is None`, then in `update_group` we make sure `result.group is not None`. I think I see why this is necessary, but might be cleaner if `is_bad_group` returned `True` if `result.group is None`. Then you could remove the check both here and in `update_group`. Though I guess you might not want to log that case.

Reply:
The None checks might actually be removed after the changes from #50. I'll do that in a follow-up.

Reply:
Yep, done with a0c1505de9a578bbd689ceaf213d45883c529661.